### PR TITLE
Use SetClip in DrawClippedLine

### LINKED ISSUE #904: RenderingExtensions.DrawClippedLine need to call RenderContext.SetClip?
### RenderingExtensions.DrawClippedLine optimization?

Why RenderingExtensions.DrawClippedLine method don't call RenderContext.SetClip and only if it returns false clips polyline manually? This done in all DrawClippedXXX methods, but not in DrawClippedLine. I can make trivial pull request for this.

### PR DESCRIPTION
**Pull request by justslon**
Fixes #904.

### CODE REVIEW

File: Source/OxyPlot/Rendering/RenderContext/RenderingExtensions.cs [hunk 1]
```diff
@@ -93,6 +93,25 @@ M3    M3
                 return;
             }
 
+            if (rc.SetClip(clippingRectangle))
+            {
+                rc.DrawLine(points, stroke, strokeThickness, dashArray, lineJoin, aliased);
+                rc.ResetClip();
```
Comment (objorke) [inline Source/OxyPlot/Rendering/RenderContext/RenderingExtensions.cs]: shouldn't it pop the clip instead?

Comment (justslon) [inline Source/OxyPlot/Rendering/RenderContext/RenderingExtensions.cs]: After small research I think you are right. It's better to call SetClip in, for example, LineSeries render. It seems clipped lines coords mostly used in annotation classes for label rendering. I think this PR can broke them. I'll try search for DrawClippedLine calls where last 2 arguments (output buffer and callback) not used and surround them with SetClip/ResetClip. I try to open new PR in a couple of days.

Comment (objorke) [inline Source/OxyPlot/Rendering/RenderContext/RenderingExtensions.cs]: I change the SetClip method in the Wpf.CanvasRenderContext to:

```csharp
        public bool SetClip(OxyRect clippingRect)
        {
            if (this.clip != null)
            {
                throw new InvalidOperationException("SetClip cannot be nested.");
            }

            this.clip = this.ToRect(clippingRect);
            return true;
        }
```

this causes a large number of tests to break.

**Comment by objorke** [comment]
I could not see anything wrong in the tests/example library png export... I think we can merge this to make the code consistent, then solve #982

**Pull request merged**